x-pack/plugins/security/server/lib/authentication/authenticator.ts

```diff
@@ -4,52 +4,79 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { Request, RequestQuery } from 'hapi';
+import { Legacy } from 'kibana';
```

Comment:
We are using `Legacy` here for `Legacy.Server`, but we're importing `Request` from `hapi` instead. It appears that `Legacy` also imports HAPI's `Request` via `Legacy.Request`. What do you think about consolidatating the imports to be consistent through the authc code to either always use the import from HAPI, or always use the import from Legacy?

Reply:
That's a great idea! I didn't realize we have `Legacy.Request`. I'd go with using `Legacy.Server/Request` everywhere to not mention Hapi at all, neither in code nor in comments. I'll update files affected by this PR and will use `Legacy.*` everywhere from now on :)